### DISCUSSION TITLE
How to run fzf first, before the list of input has completed

### DISCUSSION THREAD

**Discussion by Jiehong** [junegunn/fzf#3508] [Q&A] [answered]
Hello,

Right now, I have a command using fzf, but it's slow to produce the list of elements to filter:

```
cmd_slow_to_produce_results | subcommand | fzf --select-1 --prompt='project > '
```

Because of that, nothing happens for a few seconds, before fzf is displayed with the full list of results.

Yet, if I run `fzf` to load the full list of files, fzf has a loading indicator and loads things incrementally.

I wanted to have a similar behaviour with fzf, but I couldn't find a way to run something akin to `fzf --run 'cmd_slow_to_produce_results | subcommand'` to be dropped directly in fzf, and have the loading indicator there instead.

Is there a way to do that?

**Comment by LangLangBart** [comment]
- you could try it with the `start` event
  - added with version [0.35.0](https://github.com/junegunn/fzf/blob/master/CHANGELOG.md#0350)
  - the colon (`:`) does nothing besides returning a zero exit status
    - Ref: [Bash Reference Manual ](https://www.gnu.org/software/bash/manual/bash.html#index-_003a)

```bash
: | fzf --bind "start:reload:(sleep 2 && echo hi) || :"
```

**Comment by Jiehong** [comment]
@LangLangBart : interesting.

But what's the difference with `: | fzf --bind "start:reload:(sleep 2 && echo hi)"`? it seems to be doing the same afaict.

**Reply by LangLangBart** [reply]
Do you mean the difference between these two ?

```bash
: | fzf --bind "start:reload:(sleep 2 && echo hi) || :"
```


```bash
: | fzf --bind "start:reload:(sleep 2 && echo hi)"
```


- if the command in the reload fails, it will _suppress the warning message_ and the list will be empty.
  - Ref: #1750 

---

PS: The `:` command piped into `fzf` might be redundant in this case, this should be sufficient:

```bash
fzf --bind "start:reload:(sleep 2 && echo hi) || :"
```

**Comment by Jiehong** [comment]
Thanks @LangLangBart

**Comment by junegunn** [comment, marked as answer]
Remove `--select-1` and fzf will start immediately.

The purpose of `--select-1` is to exit without rendering the interactive finder if there's only one match. To determine this, fzf needs some items and until it does, fzf blocks. This is the opposite of what you're trying to achieve.

```sh
# Doesn't block
(sleep 2; seq 100) | fzf -q 55

# fzf automatically selects 55 and exits
(sleep 2; seq 100) | fzf -q 55 --select-1

# Because there are more then one matches, fzf starts after 2 seconds
(sleep 2; seq 100) | fzf -q 5 --select-1

# Simliar to --select-1, but it starts the finder immediately
(sleep 2; seq 100) | fzf -q 55 --bind one:accept
```
